akka-stream/src/main/scala/akka/stream/scaladsl/Queue.scala

```diff
@@ -40,36 +47,60 @@ trait SourceQueue[T] {
  * This trait adds completion support to [[SourceQueue]].
  */
 trait SourceQueueWithComplete[T] extends SourceQueue[T] {
+
   /**
-   * Complete the stream normally. Use `watchCompletion` to be notified of this
+   * Completes the stream normally. Use `watchCompletion` to be notified of this
    * operation’s success.
    */
   def complete(): Unit
 
   /**
-   * Complete the stream with a failure. Use `watchCompletion` to be notified of this
+   * Completes the stream with a failure. Use `watchCompletion` to be notified of this
    * operation’s success.
    */
   def fail(ex: Throwable): Unit
 
   /**
-   * Method returns a [[Future]] that will be completed if the stream completes,
-   * or will be failed when the stage faces an internal failure or the the [[SourceQueueWithComplete.fail]] method is invoked.
+   * Method returns a [[Future]] that will be
+   * - completed if the stream completes
+   * - failed when
+   *   - the stage faces an internal failure
+   *   - the [[SourceQueueWithComplete.fail]] method is invoked.
    */
   def watchCompletion(): Future[Done]
 }
 
+object SourceQueueWithComplete {
+  implicit def QueueOps[T](f: SourceQueueWithComplete[T]): QueueOps[T] = new QueueOps[T](f)
+  final class QueueOps[T](val queue: SourceQueueWithComplete[T]) extends AnyVal {
+    def asJava: akka.stream.javadsl.SourceQueueWithComplete[T] = SourceQueueConverter.asJava(queue)
```

Comment:
why is this added as an implicit instead of just adding it to `SourceQueueWithComplete`?

Reply:
See comment above, cannot add methods to `SourceQueueWithComplete` not to break API. Any chance we can add this change to a future, major release? I don't think there is any hurry around this change anyway..